react/tic-tac-toe/src/redux/game/reducer.js

```diff
@@ -14,24 +16,16 @@ const initialSettings = {
 
 const initialState = {
   matches: [],
-  loading: false,
-  error: '',
   settings: localStorageService.getValue(actions.GAME_SETTIGNS) || initialSettings
 };
 
-function reducer(state = initialState, action) {
-  switch (action.type) {
-    case actions.MATCHES:
-      return { ...state, loading: true };
-    case actions.MATCHES_SUCCESS:
-      return { ...state, loading: false, matches: action.payload };
-    case actions.MATCHES_FAILURE:
-      return { ...state, matches: [], loading: false, error: action.payload };
-    case actions.GAME_SETTIGNS:
-      return { ...state, settings: action.payload };
-    default:
-      return state;
+const reducerDescription = {
+  primaryActions: [actions.MATCHES],
+  override: {
+    [actions.GAME_SETTINGS]: (state, action) => ({ ...state, [action.target]: action.payload })
```

Comment:
this custom effect seems pretty similar to `onReadValue` ? https://github.com/Wolox/redux-recompose/tree/master/src/effects/onReadValue :haroldharold:

Reply:
![flat,800x800,075,f](https://user-images.githubusercontent.com/3473831/61838384-66b86f00-ae5f-11e9-9468-32ca1efad086.jpg)